packages/wallet-adapter/src/providers/connectors.ts

```
@@ -55,30 +56,44 @@ export type InitResult = Result<
+  principal?: string;
+  constructor(config: Config, meta: Meta) {
+    this.config = config;
+    this.meta = meta;
+  }
+
+  abstract init(): Promise<InitResult>;
+  abstract isConnected(): Promise<boolean>;
+  abstract createActor<Service>(
     canisterId: string,
     interfaceFactory: IDL.InterfaceFactory,
     config?: {}
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)